File: backend/app/models/clustering.py
# SPDX-FileCopyrightText: 2026 Dong Lab, Yale School of Medicine <https://donglab.org>
#
# SPDX-License-Identifier: Apache-2.0

from pydantic import BaseModel


class ClusterNode(BaseModel):
    method: str
    node_id: int
    parent_id: int | None = None
    branch_length: float
    gene_id: str | None = None
    symbol: str | None = None
    family: str | None = None

File: web/src/types/clustering.ts
// SPDX-FileCopyrightText: 2026 Dong Lab, Yale School of Medicine <https://donglab.org>
//
// SPDX-License-Identifier: Apache-2.0

export interface ClusterNode {
  method: string
  node_id: number
  parent_id: number | null
  branch_length: number
  gene_id: string | null
  symbol: string | null
  family: string | null
}
